saleor/graphql/context.py

```diff
@@ -16,18 +17,23 @@ def get_context_value(request):
     return request
 
 
-def get_app(auth_token) -> Optional[App]:
-    tokens = AppToken.objects.filter(auth_token=auth_token).values("pk")
+def get_app(raw_auth_token) -> Optional[App]:
+    tokens = AppToken.objects.filter(token_last_4=raw_auth_token[-4:]).values_list(
+        "id", "auth_token"
+    )
+    token_ids = [
+        id for id, auth_token in tokens if check_password(raw_auth_token, auth_token)
+    ]
     return App.objects.filter(
-        Exists(tokens.filter(app_id=OuterRef("pk"))), is_active=True
+        Exists(tokens.filter(id__in=token_ids, app_id=OuterRef("pk"))), is_active=True
     ).first()
 
 
 def set_app_on_context(request):
     if request.path == API_PATH and not hasattr(request, "app"):
         request.app = None
         auth_token = get_token_from_request(request)
-        if auth_token and len(auth_token) == 30:
+        if auth_token and len(auth_token) <= AppToken.auth_token.field.max_length:
```

Comment:
@korycins Could you confirm that the change in condition is safe?
As far as I remember token shorter than 30 char is a token from external authorization.  

@IKarbowiak Did you test it with external auth?

Reply:
I reverse this change. It was not needed.